
# This is a patch for CLI-Dispatch-0.16.orig to update it to CLI-Dispatch-0.16
# 
# To apply this patch:
# STEP 1: Chdir to the source directory.
# STEP 2: Run the 'applypatch' program with this patch file as input.
#
# If you do not have 'applypatch', it is part of the 'makepatch' package
# that you can fetch from the Comprehensive Perl Archive Network:
# http://www.perl.com/CPAN/authors/Johan_Vromans/makepatch-x.y.tar.gz
# In the above URL, 'x' should be 2 or higher.
#
# To apply this patch without the use of 'applypatch':
# STEP 1: Chdir to the source directory.
# STEP 2: Run the 'patch' program with this file as input.
#
#### End of Preamble ####

#### Patch data follows ####
diff -c 'CLI-Dispatch-0.16.orig/lib/CLI/Dispatch/Help.pm' 'CLI-Dispatch-0.16/lib/CLI/Dispatch/Help.pm'
Index: ./lib/CLI/Dispatch/Help.pm
*** ./lib/CLI/Dispatch/Help.pm	Sat Nov  5 03:56:57 2011
--- ./lib/CLI/Dispatch/Help.pm	Tue Jan 29 14:12:26 2013
***************
*** 86,94 ****
      foreach my $path ( @paths ) {
        my $dir = dir( $inc, $path );
        next unless $dir->exists;
!       $dir->recurse( callback => sub {
!         my $file = shift;
!         return if $file->is_dir;
  
          my $basename = $file->basename;
             $basename =~ s/\.(?:pm|pod)$//;
--- 86,95 ----
      foreach my $path ( @paths ) {
        my $dir = dir( $inc, $path );
        next unless $dir->exists;
! 
!       for my $file ( $dir->children ) {
! 
!         next if $file->is_dir;
  
          my $basename = $file->basename;
             $basename =~ s/\.(?:pm|pod)$//;
***************
*** 99,105 ****
          $class .= '::'.$basename;
  
          # ignore base class
!         return if $class eq 'CLI::Dispatch::Command';
  
          my $podfile = $file->parent->file($basename . '.pod');
          my $pmfile  = $file->parent->file($basename . '.pm');
--- 100,106 ----
          $class .= '::'.$basename;
  
          # ignore base class
!         next if $class eq 'CLI::Dispatch::Command';
  
          my $podfile = $file->parent->file($basename . '.pod');
          my $pmfile  = $file->parent->file($basename . '.pm');
***************
*** 134,140 ****
  
          my $len = length $basename;
          $maxlength = $len if $maxlength < $len;
!       });
      }
    }
  
--- 135,141 ----
  
          my $len = length $basename;
          $maxlength = $len if $maxlength < $len;
!       }
      }
    }
  
#### End of Patch data ####

#### ApplyPatch data follows ####
# Data version        : 1.0
# Date generated      : Tue Jan 29 14:16:37 2013
# Generated by        : makepatch 2.04
# Recurse directories : Yes
# Excluded files      : (\A|/).*\~\Z
#                       (\A|/).*\.a\Z
#                       (\A|/).*\.bak\Z
#                       (\A|/).*\.BAK\Z
#                       (\A|/).*\.elc\Z
#                       (\A|/).*\.exe\Z
#                       (\A|/).*\.gz\Z
#                       (\A|/).*\.ln\Z
#                       (\A|/).*\.o\Z
#                       (\A|/).*\.obj\Z
#                       (\A|/).*\.olb\Z
#                       (\A|/).*\.old\Z
#                       (\A|/).*\.orig\Z
#                       (\A|/).*\.rej\Z
#                       (\A|/).*\.so\Z
#                       (\A|/).*\.Z\Z
#                       (\A|/)\.del\-.*\Z
#                       (\A|/)\.make\.state\Z
#                       (\A|/)\.nse_depinfo\Z
#                       (\A|/)core\Z
#                       (\A|/)tags\Z
#                       (\A|/)TAGS\Z
# p 'lib/CLI/Dispatch/Help.pm' 8369 1359486746 0100644
#### End of ApplyPatch data ####

#### End of Patch kit [created: Tue Jan 29 14:16:37 2013] ####
#### Patch checksum: 94 2987 54794 ####
#### Checksum: 112 3687 47145 ####
